### PR TITLE
Prevent page from crashing if no start and end date set

### LINKED ISSUE #3532: Can't visit an unconfirmed competition's page with nil start/end dates
I noticed this while working on #3531.
Creating an unconfirmed competition with nil dates does work, but when rendering some of its views it may go at least through these methods:
  - [in_progress?](https://github.com/thewca/worldcubeassociation.org/blob/281f7b0e53501373011b9bbf6292ada159148cf4/WcaOnRails/app/models/competition.rb#L771) (which use a range with competition's dates)
  - [the schedule tab](https://github.com/thewca/worldcubeassociation.org/blob/281f7b0e53501373011b9bbf6292ada159148cf4/WcaOnRails/app/views/competitions/_competition_schedule_for_venue_table.html.erb#L8)

We should probably check these dates are not nil, or simply not allow competitions without dates :)

### PR DESCRIPTION
**Pull request by ligio90**
Fixes #3532

Screenshot from 2019-04-06 02-36-58

### CODE REVIEW

File: WcaOnRails/app/models/competition.rb [hunk 1]
```diff
@@ -461,6 +461,14 @@ def unpack_delegate_organizer_ids
     end
   end
 
+  def has_defined_dates?
+    if self.start_date.present? && self.end_date.present?
```
Comment (jfly) [inline WcaOnRails/app/models/competition.rb]: I think this can be rewritten as just `return self.start_date.present? && self.end_date.present?`

**Comment by ligio90** [comment]
Okay, i also added the email toggle thing (fixes #3934)
Screenshot from 2019-04-06 03-56-35

**Comment by ligio90** [comment]
so appearantly the return is redundant and makes the travis-ci build fail

**Comment by jonatanklosko** [comment]
Yeah, ruby functions implicitly return the value of the last line, so you can just omit `return` in that case. We have a code style check (rubocop) that enforces some good rules/practices, that's why Travis fails =)

**Comment by ligio90** [comment]
will fix that!

**Comment by ligio90** [comment]
fixed.

**Comment by jfly** [comment]
> Okay, i also added the email toggle thing (fixes #3934)

Thanks, @ligio90! Can you please split that work out into a separate PR?

**Comment by viroulep** [comment]
With or without the code for the show/hide button, the changes LGTM!

**Comment by AlbertoPdRF** [comment]
Given that it has a LGTM, are you fine merging this up although the two commits haven't been separated into two PRs @jfly?

**Comment by AlbertoPdRF** [comment]
I'll go ahead and merge this PR in ~8 hours unless someone opposes.

**Pull request merged**